### PR TITLE
generate case-based app data sources form exports

### PR DESCRIPTION
**Pull request by czue**
this is an equivalent PR to https://github.com/dimagi/commcare-hq/pull/19987 but for case data sources.

If you review by commit you'll get the bonus of discovering a bug in exports that's existed for over a year :upside_down_face: 

@proteusvacuum @emord @nickpell @esoergel

### CODE REVIEW

File: corehq/apps/export/system_properties.py [hunk 1]
```diff
@@ -355,7 +355,7 @@ def get_case_name_column(case_id_export_item):
     ExportColumn(
         tags=[PROPERTY_TAG_INFO],
         label='server_last_modified_date',
-        item=ExportItem(path=[PathNode(name='server_modified_on')], transform=USERNAME_TRANSFORM),
+        item=ExportItem(path=[PathNode(name='server_modified_on')], datatype=DATA_TYPE_DATETIME),
```
Comment (nickpell) [inline corehq/apps/export/system_properties.py]: 👍

**Review by nickpell — APPROVED** [review]

File: corehq/apps/userreports/app_manager/helpers.py [hunk 2]
```diff
@@ -24,16 +19,26 @@ def get_case_data_sources(app):
...
 def get_case_data_source(app, case_type):
-    prop_map = get_case_properties(app, [case_type], defaults=list(DEFAULT_CASE_PROPERTY_DATATYPES))
+    schema = CaseExportDataSchema.generate_schema_from_builds(
+        app.domain,
+        app._id,
+        case_type,
+        only_process_current_builds=True,
+    )
+    # the first two (row number and case id) are redundant/export specific,
+    meta_properties_to_use = MAIN_CASE_TABLE_PROPERTIES[2:]
```
Comment (proteusvacuum) [inline corehq/apps/userreports/app_manager/helpers.py]: would you want to make this a different const?

**Review by proteusvacuum — APPROVED** [review]

File: corehq/apps/userreports/app_manager/helpers.py (hunk 2 above)
Comment (czue) [reply corehq/apps/userreports/app_manager/helpers.py]: that const would be something like `CASE_TABLE_PROPERTIES_RELEVANT_TO_UCRS` or something? 

the slicing here definitely makes me feel a little icky but am unsure whether it'd be worth it to create that separately. one thing that helps with the ickiness is that tests will break if someone changes that list in a way that breaks this.

happy to do it in a follow up if you think good though. no strong feelings really.

Comment (proteusvacuum) [reply corehq/apps/userreports/app_manager/helpers.py]: :+1:

**Pull request merged**